File: weather_script.js
const fetch = require('node-fetch');

const apiKey = 'da49a559801ae670191ceab2b1f9608c'; // Replace with your OpenWeatherMap API key
const city = 'London'; // Replace with the desired city name

const getWeatherData = async (city, apiKey) => {
    const url = `https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${apiKey}&units=metric`;

    try {
        const response = await fetch(url);
        const data = await response.json();

        if (data.cod === 200) {
            const temperature = data.main.temp;
            const humidity = data.main.humidity;
            const weatherCondition = data.weather[0].description;

            console.log(`Temperature: ${temperature}°C`);
            console.log(`Humidity: ${humidity}%`);
            console.log(`Weather Condition: ${weatherCondition}`);
        } else {
            console.log(`Error: ${data.message}`);
        }
    } catch (error) {
        console.error('Error fetching weather data:', error);
    }
};

getWeatherData(city, apiKey);

File: weather_script.py
import requests

def fetch_weather(city_name, api_key):
    """
    Fetches weather data for a given city from the OpenWeatherMap API.

    Args:
        city_name (str): Name of the city.
        api_key (str): Your OpenWeatherMap API key.

    Returns:
        dict: Weather data as a dictionary, or None if request fails.
    """
    url = f"https://api.openweathermap.org/data/2.5/weather?q={city_name}&appid={api_key}&units=metric"
    try:
        response = requests.get(url)
        response.raise_for_status()
        return response.json()
    except requests.RequestException as e:
        print(f"Error fetching weather data: {e}")
        return None

if __name__ == "__main__":
    city = input("Enter city name: ")
    api_key = input("Enter your OpenWeatherMap API key: ")
    weather = fetch_weather(city, api_key)
    if weather:
        print(f"Weather in {city}: {weather['weather'][0]['description'].capitalize()}, Temperature: {weather['main']['temp']}°C")
    else:
        print("Failed to fetch weather data.")